scripts/perftune.py

```diff
@@ -549,13 +552,48 @@ def _get_irqs(self):
         pass
 
 #### Private methods ############################
+    @property
+    def __ec2_metadata_base_url(self):
+        return "http://169.254.169.254/latest/"
+
+    @property
+    def __metadata_token(self):
+        """
+        Refresh IMDSv2 session token if it necessary, and return current token
+        :return: current session token
+        """
+        token_ttl = 21600
+        update_token = False
+        if not self.__metadata_token_value:
+            update_token = True
+        else:
+            time_diff = datetime.datetime.now() - self.__metadata_token_time
+            time_diff_sec = int(time_diff.total_seconds())
+            if time_diff_sec >= token_ttl - 120:
+                update_token = True
+        self.__metadata_token_time = datetime.datetime.now()
+        req = urllib.request.Request(self.__ec2_metadata_base_url + "api/token", headers={"X-aws-ec2-metadata-token-ttl-seconds": token_ttl}, method="PUT")
+        with urllib.request.urlopen(req, timeout=0.1) as res:
+            self.__metadata_token_value = res.read().decode()
```

Comment:
You seem to be unconditionally reloading token value because you don't use the value of `update_token` as I believe you intended.

Reply:
Sorry I broke the code when I move metadata token related code to __metadata_token() method.
Pushed fixed version.